contracts/asset-proxy/contracts/src/StaticCallProxy.sol

```diff
@@ -0,0 +1,207 @@
+                if gt(staticCallDataEnd, assetDataEnd) {
+                    // Revert with `Error("INVALID_STATIC_CALL_DATA_OFFSET")`
+                    mstore(0, 0x08c379a000000000000000000000000000000000000000000000000000000000)
+                    mstore(32, 0x0000002000000000000000000000000000000000000000000000000000000000)
+                    mstore(64, 0x0000001f494e56414c49445f5354415449435f43414c4c5f444154415f4f4646)
+                    mstore(96, 0x5345540000000000000000000000000000000000000000000000000000000000)
+                    revert(0, 100)
+                }
+
+                // Copy `staticCallData` into memory
+                calldatacopy(
+                    0,                              // memory can be safely overwritten from beginning
+                    staticCallDataBegin,            // start of `staticCallData`
+                    staticCallDataLen               // copy the entire `staticCallData`
+                )
+
+                // In order to find the offset to `callTarget`, we must add:
+                // assetDataOffset
+                // + 32 (assetData len)
+                // + 4 (proxyId)
+                let callTarget := and(
+                    calldataload(add(assetDataOffset, 36)),
+                    0x000000000000000000000000ffffffffffffffffffffffffffffffffffffffff
+                )
+
+                // Perform `callTarget.staticcall(staticCallData)`
+                let success := staticcall(
+                    gas,                         // forward all gas
+                    callTarget,                  // call address `callTarget`
+                    0,                           // pointer to start of input
+                    staticCallDataLen,           // length of input
+                    0,                           // start of memory can be safely overwritten
+                    0                            // don't copy output to memory
+                )
+
+                // Copy entire output to start of memory
+                let outputLen := returndatasize()
+                returndatacopy(
+                    0,                // copy to memory at 0
+                    0,                // copy from return data at 0
+                    outputLen         // copy all return data
+                )
+
+                // Revert with reason given by `callTarget` if staticcall is unsuccessful
+                if iszero(success) {
+                    revert(0, outputLen)
+                }
+
+                // Calculate hash of output
+                let callResultHash := keccak256(0, outputLen)
+
+                // In order to find the offset to `expectedCallResultHash`, we must add:
+                // assetDataOffset
+                // + 32 (assetData len)
+                // + 4 (proxyId)
+                // + 32 (callTarget)
+                // + 32 (assetDataOffset)
```

Comment:
This should be `+ 32 (staticCallDataOffset)` instead of `+ 32 (assetDataOffset)`